src/include/duckdb/storage/index.hpp

```diff
@@ -101,6 +95,8 @@ class Index {
 	bool IsForeign() {
 		return (constraint_type == IndexConstraintType::FOREIGN);
 	}
+	//! Serializes the index and returns the pair of block_id offset positions
+	virtual std::pair<idx_t, idx_t> Serialize(duckdb::MetaBlockWriter &writer);
```

Comment:
Same here, I would replace all the pairs with structs with clear naming of the elements.